docs/content/docs/wip/ingress.md

```diff
@@ -0,0 +1,427 @@
+
+## Why do you need ingress on your cluster? 
+
+Initially, when you group pods under a service, this service is only accessible inside of the cluster. One option, of course, is to use a load balancer. However, the downside of this approach is that each service would require a new hosted load balancer, meaning more consumption of cloud resources. 
+
+However, Ingress allows you to easily establish rules for traffic routing without creating several load balancers. The Ingress Controller is built using reverse proxies, allowing it to act as a load balancer. Though the Ingress Controller requires a service to expose them to services outside of the cluster, this will be the only entrypoint that outside services need to access all the internal services on your cluster, as the Ingress Controller will redirect this traffic to the internal pods as specified in the Ingress Manifest. 
+
+## What types of ingress are supported by OSM v0.8.0?
+
+Currently, OSM v0.8.0 supports http and single service Ingress. HTTPS ingress support is experimental. OSM supports one way TLS authentication to backend services.
+
+## Ingress controller compatibility
+
+Ingress in OSM is compatible with the following ingress controllers.
+- [Kubernetes Nginx Ingress Controller][2]
+- [Azure Application Gateway Ingress Controller][3]
+- [Gloo API Gateway][5]
+
+Other ingress controllers might also work as long as they use Kubernetes Ingress resource, and allow provisioning a custom root certificate for HTTPS backend server certificate validation while using HTTPS ingress.
+
+## Configuring Ingress
+
+### Enabling Ingress
+Ingress is disabled by default when OSM is installed. However, HTTPS ingress can be enabled by updating the `osm-config` ConfigMap in `osm-controller`'s namespace (`osm-system` by default).
+
+Patch the ConfigMap by setting use_https_ingress: "true".
+> Note: To make this change persistent between upgrades, see osm mesh upgrade --help.
+```bash
+# Replace osm-system with osm-controller's namespace if using a non default namespace
+kubectl patch ConfigMap osm-config -n osm-system -p '{"data":{"use_https_ingress":"true"}}' --type=merge
+```
+
+By default, OSM configures HTTP as the backend protocol for services when an ingress resource is applied with a backend service that belongs to the mesh. A mesh-wide configuration setting in OSM's `osm-config` ConfigMap enables configuring ingress with the backend protocol to be HTTPS. 
+
+### Disabling Ingress
+
+Patch the ConfigMap by setting use_https_ingress: "false".
+> Note: To make this change persistent between upgrades, see osm mesh upgrade --help.
+```bash
+# Replace osm-system with osm-controller's namespace if using a non default namespace
+kubectl patch ConfigMap osm-config -n osm-system -p '{"data":{"use_https_ingress":"false"}}' --type=merge
```

Comment:
This is not really disabling ingress, it is disabling HTTPS ingress, while doing so will implicitly enable HTTP ingress.